### DISCUSSION TITLE
Modal position incorrect

### DISCUSSION THREAD

**Discussion by sirtobiwan** [neuefische/questions#600] [Web]
**Introduction:**
the position of the modal won't display in a correct way. The position should be next to the text "annual radiation per solar panel"

![Bildschirm­foto 2023-05-10 um 16 15 48](https://github.com/neuefische/questions/assets/125971942/34cc963e-9646-4dcf-94d5-14643a43db83)

**Pull Request:**
[PR](https://github.com/sirtobiwan/Capstone-Project-Solar-Calculator/pull/22)

**What steps to resolve the issee?**
- googled it
- checked the handout for flexbox
- asked chat gpt

**Comment by t0bbl** [comment]
I think display: flex on the <div> should help with that!?

**Reply by sirtobiwan** [reply]
thanks, but I checked on that already. I guess it's something about the positioning

**Comment by actyralf** [comment]
You should separate the icon opening the `Modal` from the actual modal content (the overlay part).
If you want to have the icon immediately after the text you have to place it inside the `StyledLabel` like below.
You have to lift the toggling logic up to your `CardRadiation` component to be able to separate the icon from the `Modal` (not part of the code below).
This makes sense anyway as you are then free to use whatever button, icon etc. as the "caller" of your `Modal`.

```js
 <StyledLabel>
          <span>Annual Radiation per Solar Panel</span>
          <FontAwesomeIcon
            icon={faInfoCircle}
            onClick={() => setShowModal(!showModal)}
          />
          <Modal
            showModal={showModal}
            onClose={() => setShowModal(false)}
            modalContent={
              <>
                // ...
              </>
            }
          />
        </StyledLabel>
```

**Comment by sirtobiwan** [comment]
thanks, it worked.
![Bildschirm­foto 2023-05-10 um 18 00 45](https://github.com/neuefische/questions/assets/125971942/49167e75-12ee-4c04-bf46-1f05a0fed028)
